src/main/java/org/truffleruby/core/time/TimeNodes.java

```diff
@@ -505,6 +501,16 @@ private DynamicObject buildTime(DynamicObject timeClass, int sec, int min, int h
                 throw new UnsupportedOperationException(StringUtils.format("%s %s %s %s", isdst, isutc, utcoffset, utcoffset.getClass()));
             }
 
+            // If the user gives sec=60, just build with sec=0 and add 60 seconds.
```

Comment:
I think it's worth noting there that `ZonedDateTime.of` will throw an exception if provided with a value of `60` for the seconds component. So doing the follow-up `plusSeconds` call is to get around that issue and allow the `ZonedDateTime` instance to carry values over.

Reply:
I meant that with `java.time does not support leap seconds (checks sec < 60)`, is it not clear enough?

Reply:
It's not clear why you'd add 60 seconds back to the time instance or why that would even work when the constructor doesn't. You write "This behaves like MRI in most cases", but don't really explain what MRI does.

I think the comment is good, but is written with some implied background knowledge.

Reply:
`add 60 seconds` means we add a duration to a Time. Since java.time checks sec < 60, the constructor cannot possibly work.
How would you describe what MRI does? For me it really just ignores the seconds and add 60 seconds, going to the next minute.